src/Main.elm

```diff
@@ -1,10 +1,197 @@
+
+        newParentName name =
+            name :: newOpened.parentName
+
+        newUsedIdList id =
+            id :: newOpened.usedIdList
+
+        newBreadcrumbpName name =
+            name :: newBreadcrumb.breadCrumbName
+
+        stateOfcurrentId id =
+            case id of
+                Just _ ->
+                    True
+
+                Nothing ->
+                    False
+    in
+    case msg of
+        MsgFolderOpened id name ->
+            { model
+                | opened =
+                    { newOpened
+                        | currentId = Just id
+                        , parentId = newParentId
+                        , parentName = newParentName name
+                        , usedIdList = newUsedIdList id
+                        , state = True
+                    }
+                , breadcrumbs =
+                    { newBreadcrumb
+                        | breadCrumbName = newBreadcrumbpName name
+                        , breadCrumbId = newBreadcrumbId id
+                    }
+            }
+
+        MsgFolderClosed ->
+            { model
+                | opened =
+                    { newOpened
+                        | currentId = newCurrentId
+                        , parentId = List.head (List.drop 2 newOpened.usedIdList)
+                        , parentName = List.drop 1 newOpened.parentName
+                        , usedIdList = List.drop 1 newOpened.usedIdList
+                        , state = stateOfcurrentId newCurrentId
+                    }
+                , breadcrumbs =
+                    { newBreadcrumb
+                        | breadCrumbName = List.drop 1 newBreadcrumb.breadCrumbName
+                        , breadCrumbId = List.drop 1 newBreadcrumb.breadCrumbId
+                    }
+            }
+
+
+view : Model -> Html Msg
+view model =
+    let
+        isOpened =
+            model.opened.state
+    in
+    Html.div [ class "container" ]
+        [ Html.h1 [ class "h1" ]
+            [ Html.text "Audience Browser"
+            ]
+        , Html.div [ class "listContainer" ]
+            [ Html.div [ id "head" ]
+                [ Html.div [ class "breadcrumbsArea" ]
+                    [ Html.a [ class "breadcrumbsText" ] [ text "Home" ]
+                    , Html.ul [ class "breadcrumbsText" ] (List.map (breadCrumbs model.breadcrumbs) (List.reverse model.breadcrumbs.breadCrumbName))
+                    ]
+                , if isOpened == True then
+                    Html.button [ class "backButton", onClick MsgFolderClosed ] [ text "Go Up" ]
+
+                  else
+                    text ""
+                ]
+            , Html.ul [ class "list" ]
+                (List.map (openFolder model.opened) model.folders)
+            , Html.ul [ class "list" ]
+                (List.map (viewAudience model.opened) model.audience)
+            ]
+        ]
+
+
+openFolder : Opened -> AudFolders.AudienceFolder -> Html Msg
+openFolder opened list =
```

Comment:
Careful with naming `list` in this case is `audienceFolder` same below with `audience`. I smell some `copy/paste` here :)

Reply:
I need to get used to naming, still new to me :)